app/interactors/tasks/destroy.rb

```diff
@@ -4,8 +4,20 @@ class Destroy
 
     delegate :task, to: :context
 
+    before do
+      context.task = task
+      context.project = task.project
```

Comment:
task already has when you delegate it, and looks like you don't use project anywhere bellow

```suggestion 
```